src/zenml/pipelines/base_pipeline.py

```diff
@@ -588,3 +594,24 @@ def _read_config_steps(
                             parameter, step_name, previous_value, value
                         )
                     )
+
+    @classmethod
+    def get_runs(cls) -> List["PipelineRunView"]:
+        """Get all past runs from the associated PipelineView.
+
+        Returns:
+            A list of all past PipelineRunViews.
+        """
+        return Repository().get_pipeline(cls).runs  # type: ignore[no-any-return]
```

Comment:
Shouldn't this code have a check if the pipeline exists?
`Repository().get_pipeline(...)` might return `None` and crash here?

Reply:
Wouldn't it already fail with the correct RuntimeError from within Repository().get_pipeline(), what would be the purpose of catching and reraising the error here?

Reply:
If I understood correctly, the `RuntimeError` only happens if a wrong type/nothing is passed to get the pipeline name.
But if you use the method correctly, it might still happen that this pipeline has never been run and isn't stored in the metadata store. That's why the `get_pipeline` method returns an `Optional[PipelineView]`. It would then fail here with an error that it's unable to find the attribute `runs` on an object of type `None` is my guess